src/maps/runGraphData.ts

```diff
@@ -31,6 +31,15 @@ export const mapRunGraphArtifactResponse: MapFunction<RunGraphArtifactResponse,
   }
 }
 
+export const mapRunGraphStateResponse: MapFunction<RunGraphStateResponse, RunGraphStateEvent> = function(source) {
+  return {
+    id: source.id,
+    timestamp: this.map('string', source.timestamp, 'Date'),
+    type: source.type as StateType,
```

Comment:
Seems like if this is true the source type should just use this type. Otherwise maybe a mapper is needed?